compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -131,6 +131,13 @@ public fun KtClassOrObject.toClassReference(module: ModuleDescriptor): Psi =
 public fun FqName.toClassReferenceOrNull(module: ModuleDescriptor): ClassReference? =
   module.asAnvilModuleDescriptor().getClassReferenceOrNull(this)
 
+@Suppress("DeprecatedCallableAddReplaceWith")
+@Deprecated("Don't rely on PSI and make the code agnostic to the underlying implementation.")
+@ExperimentalAnvilApi
+public fun FqName.toClassReferencePsiOrNull(module: ModuleDescriptor): Psi? =
+  module.asAnvilModuleDescriptor()
+    .getClassReferenceOrNull(this, preferDescriptor = false) as? Psi
```

Comment:
This function is quite unfortunate and I have to look into why we must use the PSI APIs in some cases. It's also why I had to add the `preferDescriptor` parameter.